pkg/configuration/configuration.go

```diff
@@ -280,3 +291,76 @@ func (c *Config) KeyringPrefix() (string, error) {
 	}
 	return h, nil
 }
+
+func (c *Config) CheckForUpdate(out io.Writer, current string) (Meta, error) {
+	// Check if version check is disabled in configuration
+	if c.Meta.DisableVersionCheck {
+		return c.Meta, errors.New("version check disabled")
+	}
+
+	// Check if version check has already been done today
+	yesterday := time.Now().AddDate(0, 0, -1)
+	lastCheck, err := time.Parse(time.RFC3339Nano, c.Meta.LastChecked)
+	if err != nil {
+		return c.Meta, err
+	}
+	if !lastCheck.Before(yesterday) {
+		return c.Meta, errors.New("version check already done today")
+	}
+
+	// Perform version check
+	const cliReleasesURL = "https://api.github.com/repos/appgate/sdpctl/releases"
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, cliReleasesURL, nil)
+	// Write new check time to config after request is made
+	c.Meta.LastChecked = time.Now().Format(time.RFC3339Nano)
+	if err != nil {
+		return c.Meta, err
+	}
+	req.Header.Add("Accept", "application/vnd.github+json")
+	client := &http.Client{}
```

Comment:
might be worth using the HTTPClient from the factory https://github.com/appgate/sdpctl/blob/23e58b1d42673a64e6a4b9632b215895e25a73a1/pkg/factory/http.go#L26 that we we respect HTTP_PROXY for all http requests within the project.

Reply:
Yeah, I don't see any issues with it either.